src/BinSkim.Rules/PERules/BA4001.ReportPECompilerData.cs

```diff
@@ -48,14 +48,14 @@ public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext conte
 
             if (PrintHeader)
             {
-                Console.WriteLine("Target,Compiler Name,Compiler BackEnd Version,Compiler FrontEnd Version,Language,Module Name,Module Library,Hash,Error");
+                context.CompilerDataLogger.PrintHeader();
                 PrintHeader = false;
             }
 
             if (pdb == null)
             {
-                Console.Write(context.TargetUri.LocalPath + ",");
-                Console.WriteLine($",,,,,,{context?.Hashes?.Sha256},{target.PdbParseException.Message}");
+                string errorMessage = target.PdbParseException.Message;
+                context.CompilerDataLogger.WriteException(errorMessage);
```

Comment:
>CompilerDataLogger [](http://example.com/codeflow?start=24&length=18)

what about the hash data?

Reply:
the hash is constructed with the CompilerDataLogger, so we don't need to worry here.